cachito/web/api_v1.py

```diff
@@ -29,12 +33,26 @@ def get_requests():
         value exceeds configuration's CACHITO_MAX_PER_PAGE.
     :rtype: flask.Response
     """
-    # Default versbose flag to False
+    # Check if user is filtering requests by state
+    state = flask.request.args.get('state', default=None)
+    # Default verbose flag to False
     verbose = str_to_bool(flask.request.args.get('verbose', False))
     max_per_page = flask.current_app.config['CACHITO_MAX_PER_PAGE']
     # The call to `paginate` will inspect the current HTTP request for the
     # pagination parameters `page` and `per_page`.
-    pagination_query = Request.query.paginate(max_per_page=max_per_page)
+    if state:
+        state_int = RequestStateMapping.__members__[state].value
```

Comment:
Could you add code that validates the `state` that was provided? This currently would produce a `KeyError` and provide a 500 error with no explanation to the user if the user put in an invalid state.

One of the special exceptions we use is `ValidationError`, which will return a 400 error to the user in JSON format.

Reply:
@mprahl Should we just go to the else clause if the state is invalid instead of throwing an error?

Reply:
@sarah256 I would prefer an error message over silently failing. It makes it easier to debug issues from the consumer perspective.